soroban-env-host/src/host.rs

```diff
@@ -675,6 +718,16 @@ impl Host {
             ScObject::AccountId(account_id) => {
                 self.add_host_object(account_id.metered_clone(&self.0.budget)?)
             }
+            ScObject::NonceKey(_) => {
+                Err(self.err_general("nonce keys aren't allowed to be used directly"))
+            }
+            ScObject::Account(acc) => self.add_host_object(
+                self.0
+                    .authorization_manager
+                    .borrow_mut()
+                    .add_account(self, acc.metered_clone(&self.0.budget)?)?,
```

Comment:
Where is `to_host_obj` called on an `ScObject::Account` that leads to this?

Reply:
That's just the normal code path for creating `RawVal`s out of `ScVal`s, i.e. this will get called when the function is called for the first time.